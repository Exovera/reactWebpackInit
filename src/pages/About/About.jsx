import React from 'react';
import PropTypes from 'prop-types';

const About = () => {
	return (
		<div className="about">
			hello, I'm the about page
		</div>
	);
};


About.propTypes = {

};


export default About;
