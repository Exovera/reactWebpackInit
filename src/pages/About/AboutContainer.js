import React from "react";
import { connect } from "react-redux";
import About from "./About.jsx";

// mapStateToProps is a react-redux specific function
// that injects the redux store, after filters. REQUIRED
//    state = reduxStore
//    ownProps = props passed to container Component (Not Meteor data)
const mapStateToProps = (state, ownProps) => {
	// console.log(state);
	// console.log(ownProps)
	return {
		// trends: state.base.trends,
	};
};

// // mapDispatchToProps is a react-redux specific
// // function that binds a function into the props
// // Default is: dispatch => ({ dispatch }), to this isn't necessary in simple uses
// const mapDispatchToProps = ( dispatch, ownProps ) => {
//  return {
//    open: () => {
//      dispatch( emailSuccess( true ) );
//    }
//  };
// };

// const mergeProps = ( stateProps, actionProps, parentProps ) => ({
// ...stateProps,
// ...actionProps,
// ...parentProps
// });

// Combines everything, including Meteor data
const AboutContainer = connect(
	mapStateToProps
	// mapDispatchToProps
	// mergeProps
)(About);

export default AboutContainer;