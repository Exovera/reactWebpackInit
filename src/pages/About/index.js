import AboutContainer from './AboutContainer.js';
import About from './About.jsx';

export { About };
export default AboutContainer;