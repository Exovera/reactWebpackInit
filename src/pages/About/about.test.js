import React from "react";
import Enzyme, { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import About from "./About.jsx";

Enzyme.configure({ adapter: new Adapter() });

function setup(passedProps = {}) {
	const props = {
		activeNarrative: passedProps.activeNarrative || "all",
		// activeNarrativeData: passedProps.activeNarrativeData || null,
		// filtersDisplayText: passedProps.filtersDisplayText || false,
		// scrollbarSize: passedProps.scrollPosition || 15,
		// scrollPosition: passedProps.scrollPosition || 0,
	};

	const enzymeWrapper = mount(<About {...props} />);

	return {
		props,
		enzymeWrapper,
	};
}

describe("About page", () => {
	it('should render self.', () => {
		const { enzymeWrapper } = setup();
		expect(toJSON(enzymeWrapper)).toMatchSnapshot();
	});
});