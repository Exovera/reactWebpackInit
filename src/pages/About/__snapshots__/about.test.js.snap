// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page should render self. 1`] = `
<About
  activeNarrative="all"
>
  <div
    className="about"
  >
    hello, I'm the about page
  </div>
</About>
`;
